File: Angular/Angular/ClientApp/src/app/app-core/core-controllers/authorization-api-controller.ts
import { HttpClient,HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({ providedIn: 'root' })
export class AuthorizationApiController
{

	constructor( private http: HttpClient ){}

	public isSignin(  ){
		let pars = this.toHttpParams({

		} );
		return this.http.get('/AuthorizationApi/IsSignin',pars);
	}

	public signin( Email,Password ){
		let pars = this.toHttpParams({
			Email: Email,
			Password: Password
		} );
		return this.http.get('/AuthorizationApi/Signin',pars);
	}

	public signout(  ){
		let pars = this.toHttpParams({

		} );
		return this.http.get('/AuthorizationApi/Signout',pars);
	}

	public signup( Email,Password,Confirmation,SurName,FirstName,LastName,Birthday,Tel ){
		let pars = this.toHttpParams({
			Email: Email,
			Password: Password,
			Confirmation: Confirmation,
			SurName: SurName,
			FirstName: FirstName,
			LastName: LastName,
			Birthday: Birthday,
			Tel: Tel
		} );
		return this.http.get('/AuthorizationApi/Signup',pars);
	}

	public verify(  ){
		let pars = this.toHttpParams({

		} );
		return this.http.get('/AuthorizationApi/Verify',pars);
	}

	public hasUserWithEmail( Email ){
		let pars = this.toHttpParams({
			Email: Email
		} );
		return this.http.get('/AuthorizationApi/HasUserWithEmail',pars);
	}

	public hasUserWithTel( Tel ){
		let pars = this.toHttpParams({
			Tel: Tel
		} );
		return this.http.get('/AuthorizationApi/HasUserWithTel',pars);
	}


	 toHttpParams(obj: any): {[property: string]: string} 
	 { 
	     const result: {[property: string]: string} = { }; 
	     Object.getOwnPropertyNames(obj).forEach(name => { 
	         result[name] = JSON.stringify(obj[name]); 
	     }); 
	     return result; 
	 }           
	}

File: Angular/Angular/Controllers/AuthorizationApiController.cs
using ApplicationDb.Entities;

using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;

using System;
 

namespace ASpbLibs.Application.WebAPI
{
    [ApiController]    
    [Route("api/[controller]/[action]")]
    public class AuthorizationApiController : ControllerBase
    {
        private APIAuthorization _authorization;
        private ILogger<AuthorizationApiController> _logger;

        public AuthorizationApiController(
            ILogger<AuthorizationApiController> logger,
            APIAuthorization authorization)
        {
            _authorization = authorization;
            _logger = logger;
        }

        public object IsSignin()
        {
            return new
            {
                IsSignin = _authorization.IsSignin()
            };
        }

        public object Signin(string Email, string Password)
        {
            try
            {
                _authorization.Signin(Email, Password, true);
            }
            catch (Exception ex)
            {
                return new
                {
                    status = "failed",
                    message = ex.Message
                };
            }
            return new
            {
                status = "success"
            };
        }

        public object Signout()
        {
            try
            {
                _authorization.Signout(true);
            }
            catch (Exception ex)
            {
                return new
                {
                    status = "failed",
                    message = ex.Message
                };
            }
            return new
            {
                status = "success"
            };
        } 

        
  
        public object Signup(string Email, string Password, string Confirmation, string SurName, string FirstName, string LastName, string Birthday, string Tel)
        {
            try
            {
                _authorization.Signup(Email, Password, Confirmation, SurName, FirstName, LastName, DateTime.Parse(Birthday), Tel);
            }
            catch (Exception ex)
            {
                string message = "";
                do
                {
                    message += ex.Message + ". ";
                    ex = ex.InnerException;
                }
                while (ex.InnerException != null);
                return new
                {
                    status = "failed",
                    message = message
                };
            }
            return new
            {
                status = "success"
            };
        }

        public User Verify()
        {
            return _authorization.Verify();
        }

        // 
        public object HasUserWithEmail(string Email)
        {
            bool has = _authorization.HasUserWithEmail(Email);
            if (has)
            {
                return new
                {
                    uniq = "User with this email already registered"
                };
            }
            else
            {
                return new { };
            }
            
        }

        public object HasUserWithTel(string Tel)
        {
            bool has = _authorization.HasUserWithTel(Tel);
            if (has)
            {
                return new
                {
                    uniq = "Use with this tel number already registered"
                };
            }
            else
            {
                return new { };
            } 
        }
    }



}

